src/sys/windows/udp.rs

```diff
@@ -383,6 +384,26 @@ impl Drop for UdpSocket {
     }
 }
 
+impl FromRawSocket for UdpSocket {
+    unsafe fn from_raw_socket(socket: RawSocket) -> UdpSocket {
+        let stream = net::UdpSocket::from_raw_socket(socket);
+        // Note that this cannot fail right now
+        UdpSocket::new(stream).unwrap()
```

Comment:
I think we still want to use `expect` here, since if at some point it can return an error we'll have an easier time understanding the panic.